LanguageFeatures/Patterns/shared_case_scope_A02_t01.dart

```diff
@@ -87,6 +88,19 @@ String test4(int x) {
   }
 }
 
+String test5(num? x) {
+  switch (x) {
+    case num? a? when a is Never:
+    case num? a when a != null:
+    case num? a! when a == 1:
+      a.expectStaticType<Exactly<num?>>();
```

Comment:
Wouldn't this be `num`? The first branch is a dead end, so that's ignored. The second branch promotes `a` from `num?` to `num` because we're in the true continuation of `a != null`. The third branch _also_ promotes `a` to `num` because it did not throw at `!`.

Reply:
Can this be tested at this point? It would be good to know that `a != null` actually promotes `a` in a `when` clause, but it _does_ promote in some simpler cases:

```dart
void main() {
  num? a = 1 as dynamic;
  if (a != null) a.isNegative; // OK.
}
```

Reply:
@eernstg  I've updated [shared_case_scope_A02_t01.dart](https://github.com/dart-lang/co19/pull/1832/files/ea871d40c740e654e874ea643394786acbb6b16b..ef51d98032d4266c130fededc75c2406c9607e5e#diff-ab7a3c9dc83dc25bd1f27978bdd65d5f656edcd15132e6b307bd929eec96faef) to expect `num` instead of `num?`, and added "merged at join points" test. Both tests (updated and the newly added) fail now in analyzer

Testing that `...  when a != null` actually promotes `a` I'll add later together with other type promotion tests in another PR

Reply:
Sounds good!